File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.cpp
#include <string>
#include <vector>
#include <queue>
using namespace std;


int solution(vector<int> v, int k) {
    int answer = 0;

    priority_queue<int, vector<int>, greater<int>> q(v.begin(), v.end());

    while (q.top() < k && q.size() >= 2) {
        int i = q.top();
        q.pop();
        int j = q.top();
        q.pop();

        q.push(i + j * 2);
        answer++;
    }

    return q.size() < 2 && q.top() < k ? -1 : answer;
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] arr, int K) {
        int answer = 0;
        
        Arrays.sort(arr);
        
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        
        for (int food : arr) {
            pq.offer(food);
        }
        
        if (pq.peek() >= K) return 0;
        
        while (pq.size() >= 2) {
            int first = pq.poll();
            int second = pq.poll();
            pq.offer(first + second * 2);
            answer++;
            
            if (pq.peek() >= K) return answer;
        }
        
        return -1;
    }
}